### ISSUE TITLE
[BUG] AttributeError: 'NoneType' object has no attribute 'span'

### ISSUE THREAD

**Issue by AlymAbd** [pytube/pytube#1932]
> File "app/workers.py", line 69, in download_track
>     stream = stream.streams.filter(only_audio=True).first()
>              ^^^^^^^^^^^^^^
>   File "env0/lib64/python3.12/site-packages/pytube/__main__.py", line 296, in streams
>     return StreamQuery(self.fmt_streams)
>                        ^^^^^^^^^^^^^^^^
>   File "env0/lib64/python3.12/site-packages/pytube/__main__.py", line 181, in fmt_streams
>     extract.apply_signature(stream_manifest, self.vid_info, self.js)
>   File "env0/lib64/python3.12/site-packages/pytube/extract.py", line 409, in apply_signature
>     cipher = Cipher(js=js)
>              ^^^^^^^^^^^^^
>   File "env0/lib64/python3.12/site-packages/pytube/cipher.py", line 43, in __init__
>     self.throttling_plan = get_throttling_plan(js)
>                            ^^^^^^^^^^^^^^^^^^^^^^^
>   File "env0/lib64/python3.12/site-packages/pytube/cipher.py", line 411, in get_throttling_plan
>     transform_plan_raw = find_object_from_startpoint(raw_code, match.span()[1] - 1)
>                                                                ^^^^^^^^^^
> AttributeError: 'NoneType' object has no attribute 'span'

code:
```
          search = Search("audio to find")
          results = search.results
          result = results[0].watch_url
          stream = YouTube(result)
          stream = stream.streams.filter(only_audio=True).first()
          stream.download(tempfile.gettempdir(), 'temp.mp3')

```

version: "15.0.0"
Python 3.12.3

worked yesterday (21.05.24) but now don't

like alternative i use this:

```
    stream = YouTube(result)
    stream = [x for x in stream.streaming_data['adaptiveFormats']
              if 'audio/mp4' in x['mimeType'] and 'MEDIUM' in x['audioQuality']][0]
  
    req = requests.get(stream['url'])
    with open(temp_path, 'wb') as file:
        file.write(req.content)


```

**Comment by talkinglim** [comment]
it was working 6 hours ago then it stop suddenly. Please fix.

**Comment by veered** [comment]
Looks like [pytubefix](https://github.com/juanbindez/pytubefix) (a fork of pytube) still works.

**Comment by talkinglim** [comment]
i tried pytubefix but it doesnt work with my existing code for some reason. The runtime just stop

**Comment by markjkaem** [comment]
I have the same problem.. wasted 6 hours of coding. f my life

**Comment by AlymAbd** [comment]
> I have the same problem.. wasted 6 hours of coding. f my life

try to parse streams manually (from streaming_data) and then download what you need until somebody fix it

**Comment by AlymAbd** [comment]
I incidentally closed ticket -_-, sorry

**Comment by JCPetrelli** [comment]
Same for me. It might have to do with the regex of the 'get_throttling_plan(js)' function. The regex needs to be updated because Youtube changed something ... ?

**Comment by talkinglim** [comment]
[#1805](https://github.com/pytube/pytube/pull/1805), i've tried look into this but im not good enough to spot the error and implement changes for this issue.

**Comment by markjkaem** [comment]
> Same for me. It might have to do with the regex of the 'get_throttling_plan(js)' function. The regex needs to be updated because Youtube changed something ... ?

Yes that's  what I'm thinking too. 
in cypher.py (row 405 - 412):
```
 raw_code = get_throttling_function_code(js)
    print(raw_code)
    transform_start = r"try{"
    plan_regex = re.compile(transform_start)
    print(plan_regex)
    match = plan_regex.search(raw_code)
    print(match)
    transform_plan_raw = find_object_from_startpoint(raw_code, match.span()[1] - 1)
```

This is my starting point from the raw code i printed in the console, and try{ is not in there
Vla=function(a){var b=a.split(""),c=

Seems like the raw code is different than usual?
Match is printing none

**Comment by JCPetrelli** [comment]
> > Same for me. It might have to do with the regex of the 'get_throttling_plan(js)' function. The regex needs to be updated because Youtube changed something ... ?
> 
> Yes that's what I'm thinking too. in cypher.py (row 405 - 412):
> 
> ```
>  raw_code = get_throttling_function_code(js)
>     print(raw_code)
>     transform_start = r"try{"
>     plan_regex = re.compile(transform_start)
>     print(plan_regex)
>     match = plan_regex.search(raw_code)
>     print(match)
>     transform_plan_raw = find_object_from_startpoint(raw_code, match.span()[1] - 1)
> ```
> 
> This is my starting point from the raw code i printed in the console, and try{ is not in there Vla=function(a){var b=a.split(""),c=
> 
> Seems like the raw code is different than usual? Match is printing none

yeah, i was checking the raw code from youtube but its obfuscated. But I found a "signatureCipher" function... Here it is:

... DQa=function(a,b,c){if("cmo"===b){var d;0<=(d=c.indexOf("="))?(b="cmo="+c.slice(0,d),c=c.slice(d+1)):0<=(d=c.indexOf("%3D"))&&(b="cmo="+c.slice(0,d),c=c.slice(d+3))}a[b]=c};
xP=function(a){var b=g.V(a,EQa)||a.signatureCipher;a={zY:!1,GD:"",ZG:"",s:""};if(!b)return a;b=NA(b);a.zY=!0;a.GD=b.url;a.ZG=b.sp;a.s=b.s;return a};
yP=function(a,b,c,d,e,f,h,l,m){this.Pa=a;this.startTime=b;this.duration=c;this.ingestionTime=d;this.sourceURL=e;this.Vg=m;this.endTime=b+c;this.j=h||0;this.range=f||null;this.pending=l||!1;this.Vg=m||null};
g.zP=function(){this.segments=[];this.j=null;this.B=!0};
...

I just stopped here though. If the regex needs to be rewritten this could be a starting point. Or it could also be that i'm heading in a completely wrong direction :D

**Comment by metalshanked** [comment]
This solution worked for me --> https://github.com/pytube/pytube/issues/1499

`cipher.py`

Change
`transform_plan_raw = find_object_from_startpoint(raw_code, match.span()[1] - 1)`
To
 `transform_plan_raw = js`

**Comment by OptixSec** [comment]
> This solution worked for me --> #1499
> 
> `cipher.py`
> 
> Change `transform_plan_raw = find_object_from_startpoint(raw_code, match.span()[1] - 1)` To `transform_plan_raw = js`

Thanks! That worked. Hope the main pytube package will be fixed and updated though.

**Comment by Nacnud88** [comment]
> This solution worked for me --> #1499
> 
> `cipher.py`
> 
> Change `transform_plan_raw = find_object_from_startpoint(raw_code, match.span()[1] - 1)` To `transform_plan_raw = js`

This worked. Thank you!
